fhd_core/calibration/vis_calibrate.pro

```diff
@@ -57,282 +59,234 @@ FUNCTION vis_calibrate,vis_ptr,cal,obs,status_str,psf,params,jones,vis_weight_pt
-      
-      ;Taking tile information and cross-matching it with the nonflagged tiles array, resulting in nonflagged tile arrays
-      ;grouped by cable length
-      cable_length_ref=cable_len[Uniq(cable_len,Sort(cable_len))]
-      n_cable=N_Elements(cable_length_ref)
-      
-      freq_use = (*obs.baseline_info).freq_use
-      freq_use = where(freq_use)
-      
-      cal_old = getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_notimeavg_cal/calibration/'+obs.obsname+'_cal.sav','cal')
-      gain_fit = FLTARR(2,384,128)
-      for pol_i=0, 1 do for tile_i=0,127 do $
-        if (*cal_old.amp_params[pol_i,tile_i]) NE !NULL then $
-        FOR di=0L,2 DO gain_fit[pol_i,*,tile_i]+=(*cal_old.amp_params[pol_i,tile_i])[di]*findgen(n_freq)^di
-        
-      FOR cable_i=0,n_cable-1 DO begin
-        cable_tiles = where(cable_len EQ cable_length_ref[cable_i])
-        for pol_i=0,1 do begin
-        
-          ;mean_cal = FLTARR(N_elements(freq_use))
-          ;for freq_i=0, N_elements(freq_use)-1 do mean_cal[freq_i] = mean(abs((*cal.gain[pol_i])[freq_use[freq_i],cable_tiles]))
-          ;mean_cal = mean(abs(gain[freq_use,cable_tiles]),dim=2)
-          ;fit_params=poly_fit(freq_use,mean_cal,2)
-          ;gain_fit=fltarr(n_freq)
-          ;FOR di=0L,2 DO gain_fit+=fit_params[di]*findgen(n_freq)^di
-        
-        
-          (*cal.gain[pol_i])[*,cable_tiles] = rebin(reform(abs(cal_selected[pol_i,subb,*,cable_i])),384,N_elements(cable_tiles))/mean(abs(cal_selected[pol_i,subb,freq_use,cable_i]))$
-            * reform(gain_fit[pol_i,*,cable_tiles]) * exp(Complex(0,1)*atan((*cal.gain[pol_i])[*,cable_tiles],/phase))
-        endfor
-      endfor
-      
-    endif
-    if debug_selected_cal EQ 2 then begin
-      ;cal_selected=getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_2013longrun/selected_cal/noave.sav','cal_sel')
-      cal_selected=getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_2013longrun/selected_cal/noave_zeroed.sav','updated_cal')
-      filename2='/nfs/eor-00/h1/nbarry/MWA/IDL_code/obs_list/Aug23.txt'
-      readcol, filename2, obs_ids, format='A', /silent
-      obs_id_current = where(obs_ids EQ obs.obsname)
-      for pol_i=0,1 do (*cal.gain[pol_i]) = reform(abs(cal_selected[pol_i,obs_id_current,*,*])) * exp(Complex(0,1)*atan((*cal.gain[pol_i]),/phase))
+    if keyword_set(debug_selected_cal) then begin
+        ;cal_selected=getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_2013longrun/selected_cal/noave.sav','cal_sel')
+        cal_selected=getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_2013longrun/selected_cal/noave_zeroed.sav','updated_cal')
+        filename2='/nfs/eor-00/h1/nbarry/MWA/IDL_code/obs_list/Aug23.txt'
+        readcol, filename2, obs_ids, format='A', /silent
+        obs_id_current = where(obs_ids EQ obs.obsname)
+        for pol_i=0,1 do (*cal.gain[pol_i]) = reform(abs(cal_selected[pol_i,obs_id_current,*,*])) * exp(Complex(0,1)*atan((*cal.gain[pol_i]),/phase))
```

Comment:
Please put this in a separate function if it's still needed. There shouldn't be any hardcoded paths in the main body of the code!

Reply:
old hack anyways. removed